source/extensions/filters/common/rbac/matchers.cc

```diff
@@ -224,28 +260,28 @@ IPMatcher::IPMatcher(std::unique_ptr<Network::LcTrie::LcTrie<bool>> trie, Type t
 
 bool IPMatcher::matches(const Network::Connection& connection, const Envoy::Http::RequestHeaderMap&,
                         const StreamInfo::StreamInfo& info) const {
-  const auto ip = extractIpAddress(connection, info);
-  if (!ip) {
-    return false;
-  }
-
-  return !trie_->getData(ip).empty();
-}
-
-Network::Address::InstanceConstSharedPtr
-IPMatcher::extractIpAddress(const Network::Connection& connection,
-                            const StreamInfo::StreamInfo& info) const {
+  // Directly get address and check trie without intermediate shared_ptr copies.
+  Network::Address::InstanceConstSharedPtr ip;
```

Comment:
This comment seems like a mismatch with the behavior - it *is* making a shared_ptr copy.
I think to make it not copy you'd need to use `extractIpAddress` like it was before, but returning a reference, and you'd probably also need that function to have the capability to return a reference to a static nullptr-initialized instance for the miss path.

Reply:
Thanks for flagging it. I'll go back to using `extractIpAddress` path.